docs/reference/filter-reference.md

```diff
@@ -0,0 +1,100 @@
+# Filters
+
+Filters are used to extend Ambassador to run additional custom logic on each request. Filters are writting in Golang.
+
+## Filter Definition
+
+Filters are registered with the `Filter` custom resource definition.
+
+```yaml
+---
+apiVersion: getambassador.io/v1beta1
+kind: Filter
+metadata:
+  name: param-filter
```

Comment:
I think it's worth-while to note in a comment that this is the name used in the FilterPolicy.